### PR TITLE
OreGlob update

### PR DESCRIPTION
**Pull request by Tictim**
## What
This PR contains numerous changes and internal refactors in both API and implementation side of ore expression code. Additionally this PR adds support for localization.

## Implementation Details

* All code in oreglob package was updated to use modern java syntax.
* Separated out 'known match types' into its own node instead of deriving it based on node type and property. This design change eliminates various oddities previously visible in visualization, for example identical node sometimes displaying `0 or more characters` and sometimes displaying `anything` depending on context.
* Implemented complete localization support. Both compilation report and preview are fully translatable. To preserve benefits of dynamic formatting, expression preview lang entries use domain-specific XML schema to provide visualization hints.
* Moved `OreGlob.Visualizer`, `OreGlob.VisualizationHint` into its own file. `OreGlob.Visualizer` was completely rewritten, along with its two subclasses.
* Changed major portion of `OreGlobNode` and its subclasses.
* Turned all subclasses of `OreGlobNode`s into package-private.

## Potential Compatibility Issues
Users interacting directly with `OreGlob.Visualizer` and `OreGlob.VisualizationHint` will need to update the usage.

### CODE REVIEW

File: src/main/java/gregtech/api/util/oreglob/OreGlob.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public final boolean matches(ItemStack stack) {
      * @see OreGlob#toFormattedString(String)
      */
     public final List<String> toFormattedString() {
-        return toFormattedString("  ");
+        return visualize(new OreGlobFormattedStringVisualizer()).getLines();
```
[the remark this answers is not in the document]
Comment (Tictim) [reply src/main/java/gregtech/api/util/oreglob/OreGlob.java:85]: `OreGlobVisualizer` is a builder class that produces list of strings, so I don't think reusing the instance would bring any benefits. It's still a confusing name for string list builder so I will try to come up with a better name.

File: src/test/java/gregtech/Bootstrap.java [hunk 2]
```diff
@@ -60,6 +64,8 @@ public static void perform() {
         Loader.instance().setupTestHarness(new DummyModContainer(meta));
         GregTechAPI.moduleManager = ModuleManager.getInstance();
 
+        // loadGregtechLangFile();
```
[the remark this answers is not in the document]
Comment (Tictim) [reply src/test/java/gregtech/Bootstrap.java:67]: It loads gregtech's en_us.lang file to be used on test cases. On majority of cases it's not needed, but some tests benefit from having translation available. (by "some tests" I mean oreglob test lol)

It probably should be controlled with VM argument but I don't know how to do that

File: src/test/java/gregtech/Bootstrap.java [hunk 3]
```diff
@@ -71,6 +77,18 @@ public static void perform() {
         bootstrapped = true;
     }
 
+    private static void loadGregtechLangFile() {
+        try (InputStream is = LanguageMap.class.getResourceAsStream("/assets/gregtech/lang/en_us.lang")) {
```
[the remark this answers is not in the document]
Comment (Tictim) [reply src/test/java/gregtech/Bootstrap.java:81]: This code is for tests only, so it should be fine I think.

File: src/test/java/gregtech/Bootstrap.java [hunk 4]
```diff
@@ -67,12 +71,26 @@ public static void perform() {
         managerInternal.closeRegistries();
         managerInternal.freezeRegistries();
 
+        // loadGregtechLangFile();
```
Comment (TechLord22) [inline src/test/java/gregtech/Bootstrap.java:74]: Does this need to be un-commented?

**Review by TechLord22 — APPROVED** [review]

**Pull request merged**